packages/live-preview-sdk/src/rest/entities.ts

```diff
@@ -36,30 +36,27 @@ async function updateRef(
   locale: string,
   entityReferenceMap: EntityReferenceMap,
   depth: number,
-  visitedReferences: Set<string>
+  visitedReferenceMap: Map<string, Reference>
 ): Promise<Reference | undefined | null> {
   let reference;
+  const id =
+    'urn' in updateFromEntryEditor.sys
+      ? updateFromEntryEditor.sys.urn
+      : updateFromEntryEditor.sys.id;
+
   // If the ID of the updateFromEntryEditor is in visitedReferences, then stop the recursion
-  if (visitedReferences.has(updateFromEntryEditor.sys.id)) {
+  if (visitedReferenceMap.has(id)) {
     debug.warn('Detected a circular reference, stopping recursion');
-    reference = entityReferenceMap.get(
-      'urn' in updateFromEntryEditor.sys
-        ? updateFromEntryEditor.sys.urn
-        : updateFromEntryEditor.sys.id
-    );
+    reference = visitedReferenceMap.get(id);
```

Comment:
Do we now have two maps, one for the entityReferenceMap and one visitedReferenceMap? Can't we just have one?